aws/private/py_lambda.bzl

```diff
@@ -15,6 +16,23 @@ def _short_path(file_):
         short_path = short_path[second_slash + 1:]
     return short_path
 
+
+def _get_mtree_dirs_list_to_file(path):
+    results = []
+    if "/" not in path:
+        return results
+
+    parts = ""
+    for dir_part in path.split("/")[:-1]:
+        if len(parts) == 0:
+            parts += dir_part
+        else:
+            parts += "/" + dir_part
+
+        results.append(_mtree_line(parts, type = "dir", mode = "0755"))
+
+    return results
+
 # Copied from aspect-bazel-lib/lib/private/tar.bzl
```

Comment:
Looking at this comment and your the code you are adding, is this something we could simplify by using some rules from https://github.com/bazel-contrib/tar.bzl/blob/main/docs/mtree.md?

Reply:
Didn't look too much into the details. I just saw this comment and would have guessed that this could be implemented already somewhere else. Nevermind.

Reply:
I agree we should try to de-duplicate these if we can.

Reply:
Yeah that's the obvious answer. However I suspect there's a somewhat more substantial code sharing possible here, since https://github.com/aspect-build/rules_py/blob/main/py/private/py_image_layer.bzl was created after this rule is sketched out. @thesayyn can confirm - or if you have a few minutes to compare the implementations, maybe it's obvious how to do it.